### PR TITLE
Pb99u069/datum examples

### PR DESCRIPTION
**Pull request by pb99u069**
Closes # .

### Pre-review checklist

- [x] All code has been formatted using our config (`make format`)
- [x] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - `Ctl.Examples.PlutusV2.OneShotMinting` example, demonstrating the use of `applyArgs` with a PlutusV2 script ([#1038](https://github.com/Plutonomicon/cardano-transaction-lib/pull/1038))
 - New `Contract.Test.Utils` assertions and checks: `assertOutputHasRefScript`, `checkOutputHasRefScript`, `checkTxHasMetadata` ([#1044](https://github.com/Plutonomicon/cardano-transaction-lib/pull/1044))
 - `Parallel` instance to `Contract` monad. Parallel capabilities are in the associated `ParContract` datatype ([#1037](https://github.com/Plutonomicon/cardano-transaction-lib/issues/1037))
+- `Ctl.Examples.IncludeDatum` example, demonstrating the use of `mustIncludeDatum` constraint ([#1087](https://github.com/Plutonomicon/cardano-transaction-lib/pull/1087))
```
Comment (klntsky) [inline CHANGELOG.md:41]: Examples are not a part of public API. No need to mention in the changelog.

**Review by klntsky — CHANGES_REQUESTED** [review]

File: examples/IncludeDatum.purs [hunk 2]
```diff
@@ -1,3 +1,7 @@
+-- | this module creates a transaction
+-- | that pays 2 Ada to the `IncludeDatum` script address
+-- | and then spends the script Utxo. The script only checks
+-- | that the value of the datum is equal to 42
```
Comment (klntsky) [inline examples/IncludeDatum.purs:4]:
```suggestion
-- | This module creates a transaction
-- | that pays 2 Ada to the `IncludeDatum` script address
-- | and then spends the script Utxo. The script only checks
-- | that the value of the datum is equal to 42.
```

File: examples/IncludeDatum.purs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+buildBalanceSignAndSubmitTx lookups constraints = do
+  ubTx <- liftedE $ Lookups.mkUnbalancedTx lookups constraints
+  bsTx <- liftedE $ balanceAndSignTxE ubTx
+  txId <- submit bsTx
+  logInfo' $ "Tx ID: " <> show txId
+  pure txId
+
+foreign import includeDatum :: String
+
+includeDatumScript :: Contract () Validator
```
Comment (klntsky) [inline examples/IncludeDatum.purs:118]: Please add a comment about the script semantics to this function. Also, let's rename 
```suggestion
only42Script :: Contract () Validator
```

**Review by klntsky — CHANGES_REQUESTED** [review]

File: examples/IncludeDatum.purs (hunk 3 above)
Comment (pb99u069) [reply examples/IncludeDatum.purs]: saw that there is an example named redeemer1, so I renamed mine to datum42

Comment (klntsky) [reply examples/IncludeDatum.purs]: Let's only rename the script itself. Because the example name should be descriptive. Could you please revert

**Review by klntsky — CHANGES_REQUESTED** [review]
See comment above

**Pull request merged**